---
title: "Explorative and functional data analysis"
author: "Mira Burtscher"
output:
  html_document:
    keep_md: yes
    toc: yes
    theme: united
    code_folding: hide
editor_options: 
  chunk_output_type: console
---

# General settings

```{r setup}
knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  include = TRUE,
  cache = TRUE,
  cache.lazy = FALSE,
  eval = TRUE,
  fig.width = 4 * (1 + sqrt(5)) / 2, 
  fig.height = 4,
  dpi = 700
)
```

## Packages

```{r, message=F, warning =F, include=}
library(tidyverse)
library(ggplot2);theme_set(cowplot::theme_cowplot(font_size = 15) + theme(panel.grid.major = element_line(colour = "lightgrey", linewidth = 0.2), panel.grid.minor = element_line(colour = "lightgrey", linewidth = 0.2)))
library("reshape2")
library(ggrepel)
library(knitr)
library(Biostrings)
library(RColorBrewer)
library(here)
library(org.Hs.eg.db)
```

```{r}
mutate <- dplyr::mutate
select <- dplyr::select
group_by <- dplyr::group_by
filter <- dplyr::filter

options(ggplot2.discrete.colour= c("orange3", "darkslateblue", "darkred", "darkgreen", "darkgrey", "darkviolet"))
options(ggplot2.discrete.fill= c("orange3", "darkslateblue", "darkred", "darkgreen", "darkgrey", "darkviolet"))
```

# 1. Load data

```{r}
load("data/processed_data/2024-08-15_data_list.RData")
load("data/processed_data/2024-08-15_diff_results.RData")
```

To filter secretomics data to secreted protein we use msigdb terms

```{r}
secreted_proteins <- msigdbr::msigdbr(species = "Homo sapiens") %>% 
  filter(gs_id %in% c('M5889', 'M5885'))
```

# 2. PCAs

```{r}
pca_plot <- function(data_matrix, metadata, x, y, fx, fy, l, u) {
  m <- data_matrix 
  n_obs <- nrow(m)

  m <- m[matrixStats::rowVars(m, na.rm = T) %>%
    order(decreasing = T) %>%
    head((n_obs / 100) * 10), ]

  pca <- prcomp(t(m))
  
  data <- pca$x %>%
    as.data.frame() %>%
    rownames_to_column("matrix_id") %>%
    select(matrix_id, x = {{ x }}, y = {{ y }}) %>%
    inner_join(metadata) %>%
    mutate(group = paste0(condition, "_",time))
  
  var <- (pca$sdev)^2/sum(pca$sdev^2)*100 
  vx <- as.numeric(str_extract(as.character({{ x }}) , "\\d"))
  vy <- as.numeric(str_extract(as.character({{ y }}) , "\\d"))

  p <- data %>%
    ggplot(aes(x = x * fx, y = y * fy, fill = group, shape = condition)) +
    geom_hline(yintercept = 0, colour = "lightgrey") +
    geom_vline(xintercept = 0, colour = "lightgrey") +
    geom_point(size = 4, stroke = 1, colour = "grey60") +
    scale_fill_manual(values = pal) +
    scale_shape_manual(values = c(21, 24)) +
    cowplot::panel_border() +
    coord_fixed() +
    lims(x = c(l,u), y = c(l,u)) +
    guides(fill = "none", shape = "none") +
    labs(
      x = paste0(as.character({{ x }}), " (", round(var[vx], digits = 1), "%)"),
      y = paste0(as.character({{ y }}), " (", round(var[vy], digits = 1), "%)")
    )

  plot(p)
}


p <- cowplot::plot_grid(
  pca_plot(data_list$rna$data_matrix, data_list$rna$metadata, "PC1", "PC2", 1, -1, -30, 45),
  pca_plot(data_list$proteomics$data_matrix, data_list$proteomics$metadata, "PC1", "PC2", 1, 1, -10, 15),
  pca_plot(data_list$input_phospho$data_matrix, data_list$input_phospho$metadata, "PC2", "PC3", -1, 1, -15, 12),
  pca_plot(data_list$secretomics$data_matrix, data_list$secretomics$metadata, "PC1", "PC2", 1, 1, -12, 10),
  nrow = 1
)
plot(p)
```

# 3. DE result visualisations

## Number of hits

```{r}
lfc_cutoff <- log2(1.5)
p_cutoff <- 0.05

hit_table <- diff_results %>%
   filter(!(modality == "secretomics" & !(feature_id %in% secreted_proteins$gene_symbol))) %>% 
  subset(abs(logFC) >= lfc_cutoff & adj.P.Val <= p_cutoff) %>%
  mutate(status = ifelse(logFC > 0, "up", "down")) %>% 
  group_by(modality, comparison, status, time) %>%
  summarise(count = n())

dummy_df <- data.frame(modality = c(unique(hit_table$modality)),
                       time = factor(c("0h")), 
                       count = 0,
                       status = c(rep("up", length(unique(hit_table$modality))), rep("down", length(unique(hit_table$modality)))))

hit_table %>%
  bind_rows(dummy_df) %>%
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h"))) %>%
  ungroup() %>% 
  tidyr::complete(time, status, modality, fill = list(count = 0)) %>% 
  mutate(time = as.numeric(str_remove(time,"h"))) %>% 
  filter(modality != "phospho") %>% 
  mutate(modality = ifelse(modality == "rna", "Transcriptome", modality)) %>% 
   mutate(modality = ifelse(modality == "input_phospho", "Phosphoproteome", modality)) %>% 
  mutate(modality = ifelse(modality == "proteomics", "Proteome", modality)) %>% 
  mutate(modality = ifelse(modality == "secretomics", "Secretome", modality)) %>% 
  ggplot(aes(x = time, y = count, label = count, colour = status)) +
  geom_point(size = 2.5) +
  geom_line(aes(group = status), size = 1) +
  scale_colour_manual(values = c("up" = "indianred", "down"= "dodgerblue4")) +
  #guides(x = guide_axis(angle = 60)) +
  facet_wrap(~modality, scales = "free_y", nrow =1) +
  labs(x = "time [h]", y= "# of hits")
```


## Volcano

```{r}
# volcano plot
diff_results %>% 
  mutate(hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>% 
  mutate(hit = ifelse(adj.P.Val < 0.05 & abs(logFC) < log2(2) & modality == "rna", "no hit", hit)) %>% 
  mutate(status = ifelse(hit == "hit" & logFC > 0, "up", "not affected")) %>% 
  mutate(status = ifelse(hit == "hit" & logFC < 0, "down", status)) %>% 
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h"))) %>% 
  mutate(modality = ifelse(modality == "rna", "Transcriptome", modality)) %>% 
   mutate(modality = ifelse(modality == "input_phospho", "Phosphoproteome", modality)) %>% 
  mutate(modality = ifelse(modality == "proteomics", "Proteome", modality)) %>% 
  mutate(modality = ifelse(modality == "secretomics", "Secretome", modality)) %>% 
  filter(modality != "phospho") %>% 
  ggplot(aes(x = logFC, y = -log10(P.Value), color = status)) +
  #scale_color_gradient2(low = "dodgerblue4", mid = "lightgrey", high = "indianred", midpoint = 0) +
  geom_hline(yintercept = -log10(p_cutoff), lty = 2) +
  geom_vline(xintercept = c(-lfc_cutoff, lfc_cutoff), lty = 2) +
  scale_colour_manual(values = c("up" = "indianred","not affected" ="lightgrey",  "down"= "dodgerblue4")) +
  geom_point(alpha = 0.5) +
  facet_grid(rows = vars(modality), cols = vars(time)) +
  cowplot::theme_cowplot()+
  cowplot::panel_border()

```

## Heatmap

```{r}
overlap <- diff_results %>%
  subset(abs(logFC) >= lfc_cutoff & adj.P.Val <= p_cutoff) %>%
  filter(!(modality == "rna" & abs(logFC) < log2(2))) %>% 
  filter(!(modality == "secretomics" & !(feature_id %in% secreted_proteins$gene_symbol))) %>% 
  filter(modality != "phospho" & modality != "input_phospho") %>%
  # mutate(modality = ifelse(modality == "rna", "rna", "protoemics technologies")) %>%
  # modify feature_id to be able to collapse phospho
  ungroup() %>%
  distinct(modality, feature_id)%>% 
  group_by(feature_id) %>% 
  summarise(n_hits = n_distinct(modality)) %>% 
  filter(n_hits >1) %>% 
  bind_rows(data.frame(feature_id = c("MARCKS", "CDH11", "NRP2", "SDC4", "TFPI", "CPA4", "CACHD1")))

data <- diff_results %>% 
  filter(modality != "phospho" & modality != "input_phospho") %>% 
  filter(feature_id %in% overlap$feature_id) %>% 
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h")),
          hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>%
  #filter(hit == "hit") %>% 
  acast(feature_id ~ time + modality, value.var = "logFC", fill = NaN)

data2 <- diff_results %>% 
  filter(modality != "phospho" & modality != "input_phospho") %>% 
  filter(feature_id %in% overlap$feature_id) %>% 
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h")),
          hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>%
  filter(hit == "hit") %>% 
  acast(feature_id ~ time + modality, value.var = "adj.P.Val", fill = 1)


col_order <- c(
  "12h_secretomics", "24h_secretomics", "48h_secretomics", "72h_secretomics", "96h_secretomics",
  "12h_proteomics", "24h_proteomics", "48h_proteomics", "72h_proteomics", "96h_proteomics",
  "1h_rna", "12h_rna", "24h_rna", "48h_rna", "72h_rna", "96h_rna"
  )

data <- data[,col_order]  %>% t

data2<- data2[,col_order] %>% t

data %>% 
  ComplexHeatmap::Heatmap(na_col = "white",
   # km = 3,
    rect_gp = grid::gpar(col = "white", lwd = 0.5),
  cluster_columns = T, 
  show_row_dend = F,
  show_column_dend = F,
  show_row_names = T,
   row_names_gp = grid::gpar(fontsize = 7),  
   column_names_gp = grid::gpar(fontsize = 7),  
  cell_fun = function(j, i, x, y, w, h, fill) {
    if(data2[i, j] < 0.05) {
        grid::grid.text("*", x, y)
    }
    },
  col = circlize::colorRamp2(c(-3, 0, 3), c("dodgerblue4", "grey95","darkred")),
  name = "log2FC"
  )
  
```

### phospho

```{r}
hits <- diff_results %>%filter(modality == "input_phospho") %>%  mutate( hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>%   filter(hit == "hit")%>%
  arrange(-abs(logFC)) %>% 
  head(100)

data <- diff_results %>% 
  filter(modality == "phospho") %>% 
  filter(feature_id %in% hits$feature_id) %>% 
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h")),
          hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>% 
  acast(feature_id ~ time + modality, value.var = "logFC", fill = NaN) %>% 

data2 <- diff_results %>% 
  filter(modality == "phospho") %>% 
  filter(feature_id %in% hits$feature_id) %>% 
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h")),
          hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>%
  acast(feature_id ~ time + modality, value.var = "adj.P.Val", fill = 1) %>%  t()


data %>% 
  ComplexHeatmap::Heatmap(na_col = "white",
   # km = 3,
    rect_gp = grid::gpar(col = "white", lwd = 0.5),
  cluster_columns = F, 
  show_row_dend = F,
  show_column_dend = F,
  show_row_names = T,
   row_names_gp = grid::gpar(fontsize = 6),  
   column_names_gp = grid::gpar(fontsize = 6),  
  cell_fun = function(j, i, x, y, w, h, fill) {
    if(data2[i, j] < 0.05) {
        grid::grid.text("*", x, y)
    }
    },
  col = circlize::colorRamp2(c(-3, 0, 3), c("dodgerblue4", "grey95","darkred")),
  name = "log2FC"
  )
  

```

### without stars

```{r}
hits <- diff_results %>%filter(modality == "input_phospho") %>%  mutate( hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>%   filter(hit == "hit")%>%
  arrange(-abs(logFC)) %>% 
  head(80)

data <- diff_results %>% 
  mutate( hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>% 
  filter(modality == "input_phospho" & hit == "hit") %>% 
  filter(feature_id %in% hits$feature_id) %>% 
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h")),
          hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>% 
  acast(feature_id ~ time + modality, value.var = "logFC", fill = 0)


data %>%
  ComplexHeatmap::Heatmap(
    na_col = "white",
    # km = 3,
    rect_gp = grid::gpar(col = "white", lwd = 0.5),
    cluster_columns = F,
    show_row_dend = F,
    show_column_dend = F,
    show_row_names = T,
    row_names_gp = grid::gpar(fontsize = 6),
    col = circlize::colorRamp2(c(-2, 0, 2), c("dodgerblue4", "white", "darkred")),
    name = "log2FC"
  )
  
```


## Venns

```{r}
times <- diff_results$time %>% unique()

secreted_proteins <- msigdbr::msigdbr(species = "Homo sapiens") %>% 
  filter(gs_id %in% c('M5889', 'M5885'))


input <- diff_results %>%
  subset(abs(logFC) >= lfc_cutoff & adj.P.Val <= p_cutoff) %>%
  # potentially filter for 1 timepoint
  # filter(time == "12h") %>% 
  # stronger cutoff for rna
  filter(!(modality == "rna" & abs(logFC) < 2)) %>% 
  # only secreted
  filter(!(modality == "secretomics" & !(feature_id %in% secreted_proteins$gene_symbol))) %>% 
  #filter(modality != "phospho") %>%
  ##mutate(modality = ifelse(modality == "rna", "rna", "protoemics technologies")) %>%
  # modify feature_id to be able to collapse phospho
  mutate(feature_id = str_replace_all(feature_id, "___", "_")) %>%
  separate(feature_id, into = c("feature_id", "sequence", "multiplicity", "siteID")) %>%
  ungroup() %>%
  distinct(time, modality, feature_id)



plot_venn <- function(input_list){
  p <- venn.diagram(
  x = list(
    input_list$rna  %>% select(feature_id) %>% unlist(),
    input_list$proteomics %>% select(feature_id) %>% unlist(),
    input_list$secretomics %>% select(feature_id) %>% unlist()
  ),
  filename = NULL,
  category.names = c("rna", "prot", "secretomics"),
  col = c("mediumseagreen", "#440154ff", "darkcyan"),
  fill = c(alpha("mediumseagreen", 0.5), alpha("#440154ff", 0.5),  alpha("darkcyan", 0.5)),
  cex = 0.8,
  cex.prop = NULL,
  cat.cex = 1,
  cat.default.pos = "text",
  fontfamily = "serif",
  ext.percent = .005,
  # cat.pos = c(-27, 27),
  cat.dist = c(0.055, 0.055, 0.055)
)
dev.off()
grid::grid.draw(p)
}

input_008h <- split(input %>%  filter(time == "0.08h"), input$modality)
plot_venn(input_008h)




```


# 4.MSIGDB enrichment code

```{r}
msigdf_decoupler <- msigdbr::msigdbr(species = "Homo sapiens", category = "H") %>%
  transmute(source = str_replace(gs_name, "HALLMARK_", ""), mor = 1, target = gene_symbol) %>%
  distinct()
```

```{r}
decoupler_results <- diff_results %>% 
  # only use the corrected phospho
  filter(modality != "phospho") %>% 
  filter(!(modality == "secretomics" & !(feature_id %in% secreted_proteins$gene_symbol))) %>% 
  # modify feature_id to be able to collapse phospho
  mutate(feature_id = str_replace_all(feature_id,"___", "_")) %>% 
  separate(feature_id, into =c("feature_id", "sequence", "multiplicity", "siteID")) %>% 
  mutate(absFC = abs(logFC)) %>% 
  group_by(modality,time, feature_id) %>%
  # collapse phospho
  mutate(maxlogFC = max(abs(logFC))) %>% 
  filter(maxlogFC == absFC) %>% 
  distinct(modality,time, feature_id, logFC) %>% 
  group_by(modality) %>% 
  nest(data = c(feature_id, logFC, time))%>% 
  mutate(ER = map(data, function(df) { 
    x = df %>%  acast(feature_id ~ time, value.var = "logFC")
    decoupleR::run_wmean(mat = x, network = msigdf_decoupler)
  }
)) %>% 
  select(-data) %>% 
  unnest(cols = c(ER))
```

Visualise significant pathways of all modalities and timepoints together

```{r}
decoupler_results %>% 
  filter(statistic == "norm_wmean" & p_value < 0.05 & abs(score) > 2) %>% 
  ggplot(aes(y = source, x= score, colour = modality)) +
  geom_point() +
  geom_vline(xintercept = c(-3,0,3), linetype =3) +
  facet_wrap(~condition, scales = "free_y", ncol =2) +
  theme(axis.text.y = element_text(size = 6), panel.grid.major.y = element_line(colour = "grey"))
```

## Pathway zoom in

```{r}
int <- c("EPITHELIAL_MESENCHYMAL_TRANSITION", "TGF_BETA_SIGNALING")

decoupler_results %>% 
  filter(statistic == "norm_wmean") %>% 
  filter(source %in% int) %>% 
  mutate(time = factor(condition, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h"))) %>% 
  ggplot(aes(y = source, x= score, colour = time, shape = modality, group = modality)) +
  geom_point(size =4, pos = position_dodge(width = 0.5), alpha = 0.5) +
  geom_vline(xintercept = c(-2,0,2), linetype =c(3,1,3)) +
  scale_colour_manual(values = c("#D0D1E6", "#A6BDDB", "#67A9CF", "#3690C0", "#02818A", "#016C59", "#014636")) 
  #facet_wrap(~condition, scales = "free_y", ncol =2) +
  theme(axis.text.y = element_text(size = 10), panel.grid.major.y = element_line(colour = "lightgrey"))
  
  
decoupler_results %>% 
  filter(statistic == "norm_wmean") %>% 
  filter(source %in% int) %>% 
  mutate(time = factor(condition, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h"))) %>% 
  mutate(time = as.numeric(str_remove(time, "h"))) %>% 
  ggplot(aes(y = score, x= time, colour = modality, group = paste0(modality, source))) +
  geom_point() +
  geom_line() +
  facet_wrap(~source)
```

## Save

```{r, eval=F}
save(decoupler_results, file = paste0("results/", Sys.Date(), "_pathwayenrichment_results.RData"))
```


#3. Kinase enrichment



```{r}
res_kinase_enrichment <- list()

res_kinase_enrichment$KSN  <- OmnipathR::get_signed_ptms() %>%
  dplyr::filter(modification %in% c("dephosphorylation","phosphorylation")& sources != "ProtMapper") %>%
  dplyr::filter(!(stringr::str_detect(sources, "ProtMapper") & n_resources == 1)) %>% 
  dplyr::mutate(p_site = paste0(substrate_genesymbol, "_",residue_type, residue_offset),
                mor = ifelse(modification == "phosphorylation", 1, -1),
                likelihood = 1) %>%
  dplyr::transmute(p_site, enzyme_genesymbol, mor) %>%
  as.data.frame() %>% 
  mutate( id = paste(p_site,enzyme_genesymbol,sep =""))

res_kinase_enrichment$KSN <- res_kinase_enrichment$KSN[!duplicated(res_kinase_enrichment$KSN$id),]
res_kinase_enrichment$KSN <- res_kinase_enrichment$KSN[,-5]


res_kinase_enrichment$KSN <- select(res_kinase_enrichment$KSN, source= enzyme_genesymbol, target = p_site, mor)

# Using the run_wmean of decoupleR
res_kinase_enrichment$enrichment <- 
  diff_results %>%  
  filter(modality == "input_phospho") %>% 
  mutate(siteID = str_extract(feature_id, ".\\d+$"),
         Gene = str_extract(feature_id,"^[:alnum:]+"))%>% 
  mutate(substrateID = paste0(Gene, "_", siteID)) %>% 
  group_by(time, substrateID) %>% 
  summarise(logFC = mean(logFC))%>% 
  group_by(time) %>% 
  nest(data = c(substrateID, logFC)) %>% 
  mutate(ER = map(data,function(df){
    ER = as.data.frame(
  decoupleR::run_wmean(
    mat = as.matrix(df %>% column_to_rownames("substrateID")),
    network =res_kinase_enrichment$KSN,
    .source = "source"
  ))
  }))%>%
    select(-data) %>%
    unnest()
```

## Extract results

Visualise as heatmap

```{r}
hit_kinases <- res_kinase_enrichment$enrichment %>% 
  filter(p_value < 0.03 & statistic == "norm_wmean" & abs(score) >3)
# 

res_kinase_enrichment$enrichment %>% 
  filter(source %in% hit_kinases$source& statistic == "norm_wmean") %>% 
  mutate(time = factor(time, levels=c("0.08h", "1h", "12h", "24h", "48h", "72h", "96h"))) %>% 
  acast(time~source, value.var = "score") %>% 
  t() %>% 
  #scale() %>% 
   ComplexHeatmap::Heatmap(
  cluster_columns = F,
  # number of clusters
  km = 5,
  show_row_dend = F,
  show_column_dend = F,
  show_row_names = T,
   row_names_gp = grid::gpar(fontsize = 8),  
  col = circlize::colorRamp2(c(-6, 0, 6), c("dodgerblue4", "white","darkred")),
  name = "kinase_score"
  )

```

```{r}
hit_tfs <- res_kinase_enrichment$enrichment %>% 
  filter(p_value < 0.03 & statistic == "norm_wmean" & abs(score) >3) %>% 
  arrange(abs(score)) %>%  head(10)

res_kinase_enrichment$enrichment %>% 
  filter(source %in% hit_tfs$source& statistic == "norm_wmean") %>% 
  mutate(time = factor(time, levels=c("0.08h", "1h", "12h", "24h", "48h", "72h", "96h"))) %>% 
  ggplot(aes(x = score, y = reorder(source, score), fill = time)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#D0D1E6", "#A6BDDB", "#67A9CF", "#3690C0", "#02818A", "#016C59", "#014636")) +
  geom_vline(xintercept = 0)
```

## MSIGDB Ernichment

MSigdb enrichment with kinases

```{r}
res_kinase_enrichment$ER_msig <-  res_kinase_enrichment$enrichment %>%
  filter(source %in% hit_kinases$source& statistic == "norm_wmean") %>% 
  distinct(time, source, score) %>% 
  acast(source ~ time, value.var = "score") %>% 
  decoupleR::run_wmean(network = msigdf_decoupler)

res_kinase_enrichment$ER_msig %>% 
  filter(statistic == "norm_wmean" & p_value < 0.03) %>% 
  ggplot(aes(y = source, x= score, colour = condition)) +
  geom_point() +
  geom_vline(xintercept = 0, linetype =3) +
  #facet_wrap(~condition, scales = "free_y", ncol =2) +
  theme(axis.text.y = element_text(size = 8), panel.grid.major.y = element_line(colour = "grey"))
```

## Kinase target visualisation

```{r}

kin <- "PRKCZ"

deg <- diff_results %>%
  filter(modality == "input_phospho") %>% # try with phospho as well
  mutate(siteID = str_extract(feature_id, ".\\d+$"),
         Gene = str_extract(feature_id,"^[:alnum:]+"),
         target = paste0(Gene, "_", siteID))%>%
  ungroup()%>%
  distinct(target, time, Gene, t, logFC, P.Value, adj.P.Val) %>%
  mutate(hit = ifelse(adj.P.Val < 0.05 &abs(logFC) > log(1.5), "hit", "no hit")) %>%
  mutate(hit = ifelse(hit == "hit" & logFC < 0, "down", hit)) %>%
  mutate(hit = ifelse(hit == "hit" & logFC > 0, "up", hit)) %>%
  mutate(label = ifelse(hit != "no hit", target, "")) %>%
  mutate(time = factor(time, levels = c("0.08h", "1h", "12h","24h", "48h", "72h", "96h")))


df <- res_kinase_enrichment$KSN %>%
  filter(source == kin)


pMAP2K1 <- deg %>%
  filter(target %in% df$target) %>%
  ggplot(aes(x = logFC, y = -log10(P.Value), colour = hit)) +
  geom_point() +
  geom_text_repel(aes(label = label), size =2) +
  scale_colour_manual(values = c("up" = "indianred", "no hit" = "lightgrey","down" = "dodgerblue4")) +
  facet_wrap(~time)+
  ggtitle("MET")
pMAP2K1


```

## Save

```{r}
save(res_kinase_enrichment, file ="results/2024-07-24_kinase_enrichment_result.RData")
```

# 4. TFs

```{r}
library(decoupleR)
res_tf_enrichment <- list()
res_tf_enrichment$net_dorothea <- get_dorothea(organism='human', levels=c('A', 'B', 'C')) 

res_tf_enrichment$enrichment <-   
  diff_results %>%  
  filter(modality == "rna") %>% 
  mutate(Gene =feature_id)%>% 
  ungroup() %>% 
  distinct(time, Gene, t) %>% 
  group_by(time) %>% 
  nest(data = c(Gene, t))%>% 
  mutate(ER = map(data,function(df){
    ER = as.data.frame(
  decoupleR::run_mlm(
    mat = as.matrix(df %>% column_to_rownames("Gene")),
     net=res_tf_enrichment$net_dorothea,
    .source='source',
    .target='target', 
    .mor='mor', 
    minsize = 5
  ))
  }))%>%
    select(-data) %>%
    unnest(cols = c(ER))
```

## Extract results

```{r}

hit_tfs <- res_tf_enrichment$enrichment %>% 
  filter(p_value < 0.03 & statistic == "mlm" & abs(score) >3)
# 
set.seed(1)
res_tf_enrichment$enrichment %>% 
  filter(source %in% hit_tfs$source& statistic == "mlm") %>% 
  mutate(time = factor(time, levels=c("0.08h", "1h", "12h", "24h", "48h", "72h", "96h"))) %>% 
  acast(time~source, value.var = "score") %>% 
  t() %>% 
  #scale() %>% 
   ComplexHeatmap::Heatmap(
  cluster_columns = F,
  # number of clusters
  km = 5,
  show_row_dend = F,
  show_column_dend = F,
  show_row_names = T,
   row_names_gp = grid::gpar(fontsize = 8),  
  col = circlize::colorRamp2(c(-6, 0, 6), c("dodgerblue4", "white","darkred")),
  name = "tf_score"
  )
```

```{r}
hit_tfs <- res_tf_enrichment$enrichment %>% 
  filter(p_value < 0.03 & statistic == "mlm" & abs(score) >3) %>% 
  arrange(abs(score)) %>%  head(10)

res_tf_enrichment$enrichment %>% 
  filter(source %in% hit_tfs$source& statistic == "mlm") %>% 
  mutate(time = factor(time, levels=c("0.08h", "1h", "12h", "24h", "48h", "72h", "96h"))) %>% 
  ggplot(aes(x = score, y = reorder(source, score), fill = time)) +
  geom_bar(stat = "identity") +
  scale_fill_manual(values = c("#D0D1E6", "#A6BDDB", "#67A9CF", "#3690C0", "#02818A", "#016C59", "#014636")) +
  geom_vline(xintercept = 0)
```

## TF target visualisation

```{r}

kin <- "RELA"

deg <- diff_results %>%
  filter(modality == "rna") %>% # try with phospho as well
  distinct(time, feature_id, t, logFC, P.Value, adj.P.Val) %>%
  mutate(hit = ifelse(adj.P.Val < 0.05 &abs(logFC) > log(1.5), "hit", "no hit")) %>%
  mutate(hit = ifelse(hit == "hit" & logFC < 0, "down", hit)) %>%
  mutate(hit = ifelse(hit == "hit" & logFC > 0, "up", hit)) %>%
  mutate(label = ifelse(hit != "no hit", feature_id, "")) %>%
  mutate(time = factor(time, levels = c("0.08h", "1h", "12h","24h", "48h", "72h", "96h")))


df <- res_tf_enrichment$net_dorothea %>%
  filter(source == kin)


pMAP2K1 <- deg %>%
  filter(feature_id %in% df$target) %>%
  ggplot(aes(x = logFC, y = -log10(P.Value), colour = hit)) +
  geom_point() +
  geom_vline(xintercept = 0) +
  geom_text_repel(aes(label = label), size =2) +
  scale_colour_manual(values = c("up" = "indianred", "no hit" = "lightgrey","down" = "dodgerblue4")) +
  facet_wrap(~time)+
  labs(subtitle = kin)
pMAP2K1


```

## Save

```{r, eval =F}
save(res_tf_enrichment, file = "results/2024-07-24_tf_enrichment_results.RData")
```

# 5. Enzyme expression profiles

```{r}
enzymes_df <- res_kinase_enrichment$enrichment %>% 
  filter(p_value < 0.03 & abs(score) > 3 & time %in% c("0.08h", "1h",  "12h")) %>% 
  mutate(abs_score = abs(score)) %>% 
  group_by(source) %>% 
  mutate(max_score = max(abs(score))) %>% 
  filter(max_score == abs_score) %>% 
  distinct(source, score) %>% 
  mutate(enzyme = "kinase") %>% 
  bind_rows(res_tf_enrichment$enrichment %>% 
  filter(p_value < 0.03 & abs(score) > 3 & time %in% c("0.08h", "1h",  "12h")) %>% 
  mutate(abs_score = abs(score)) %>% 
  group_by(source) %>% 
  mutate(max_score = max(abs(score))) %>% 
  filter(max_score == abs_score) %>% 
  distinct(source, score)%>% 
  mutate(enzyme = "transcription factor")) %>% 
  ungroup() %>% 
  drop_na() %>% 
  distinct(source, enzyme)

diff_results %>% 
  filter(modality %in% c("rna", "proteomics")) %>% 
  left_join(enzymes_df, by = join_by(feature_id == source)) %>% 
  mutate(enzyme = ifelse(is.na(enzyme), "other", enzyme),
          time = factor(time, levels = c("0h", "0.08h", "1h", "12h", "24h", "48h", "72h", "96h")),
         hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>% 
  mutate(label = ifelse(hit == "hit" & enzyme != "other", feature_id, "")) %>% 
  mutate(enzyme = factor(enzyme, levels = c("other", "kinase",  "transcription factor"))) %>% 
  arrange(enzyme) %>% 
  ggplot(aes(x = logFC, y = -log10(P.Value), colour = enzyme, group = enzyme)) +
  geom_vline(xintercept = 0, colour = "black") +
  geom_point() + 
  ggrepel::geom_label_repel(aes(label = label), max.overlaps = Inf) +
  facet_grid(modality~time, scales = "free") +
  scale_color_manual(values = c("grey", "darkblue", "darkgreen")) +
  cowplot::panel_border()
```

```{r}
diff_results %>% 
  filter(modality %in% c("rna", "proteomics")) %>% 
  left_join(enzymes_df, by = join_by(feature_id == source)) %>% 
  mutate(enzyme = ifelse(is.na(enzyme), "aother", enzyme),
          time = factor(time, levels = c("0h", "0.08h", "1h", "12h", "24h", "48h", "72h", "96h")),
         hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit"))  %>% 
  filter(enzyme != "aother") %>% 
  acast(feature_id ~ time + modality, value.var = "logFC", fill = 0) %>% 
   ComplexHeatmap::Heatmap(
  cluster_columns = T,
  show_row_dend = F,
  show_column_dend = F,
  show_row_names = T,
   row_names_gp = grid::gpar(fontsize = 8),  
  col = circlize::colorRamp2(c(-1, 0, 1), c("dodgerblue4", "white","darkred")),
  name = "tf_score"
  )
```
