---
title: "Compare Nadines multi-omics data to other studies"
output:
  html_document:
    keep_md: yes
    toc: yes
    theme: united
    code_folding: hide
editor_options: 
  chunk_output_type: console
---

# General settings

```{r setup}
knitr::opts_chunk$set(
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  include = TRUE,
  cache = TRUE,
  cache.lazy = FALSE,
  eval = TRUE,
  fig.width = 4 * (1 + sqrt(5)) / 2, 
  fig.height = 4,
  dpi = 700
)
knitr::opts_knit$set(root.dir = "~/Documents/01_repos/kidney_fibrosis/")
```

## Packages

```{r, message=F, warning =F, include=}
library(tidyverse)
library(ggplot2);theme_set(cowplot::theme_cowplot(font_size = 12) + theme(panel.grid.major = element_line(colour = "lightgrey", linewidth = 0.2), panel.grid.minor = element_line(colour = "lightgrey", linewidth = 0.2)))
library("reshape2")
library(ggrepel)
library(knitr)
library(Biostrings)
library(RColorBrewer)

mutate <- dplyr::mutate
select <- dplyr::select
group_by <- dplyr::group_by
```

```{r}
options(ggplot2.discrete.colour= c("orange3", "darkslateblue", "darkred", "darkgreen", "darkgrey", "darkviolet"))
options(ggplot2.discrete.fill= c("orange3", "darkslateblue", "darkred", "darkgreen", "darkgrey", "darkviolet"))
tilted <-  theme(axis.text.x = element_text(angle = 45, hjust =1, vjust = 1))

pal_condition = c(
  "TGF-\u03b2" = "#FEA873FF",
  "control" = "#D9D9D9")

pal_cond_time <- c(
  "ctrl_0h" = "#F0F0F0",
  "ctrl_0.08h" = "#D9D9D9",
  "ctrl_1h" = "#BDBDBD",
  "ctrl_12h" = "#969696",
  "ctrl_24h" = "#737373",
  "ctrl_48h" = "#525252",
  "ctrl_72h" = "#252525",
  "ctrl_96h" = "#000000",
  "TGF_0h" = "#FCFDBFFF",
  "TGF_0.08h" =  "#FED395FF",
  "TGF_1h" = "#FEA873FF",
  "TGF_12h" = "#F97C5DFF",
  "TGF_24h" = "#E95562FF",
  "TGF_48h" = "#C83E73FF" ,
  "TGF_72h" = "#A3307EFF",
  "TGF_96h" =   "#7D2482FF"
)

pal_omics <-
  c(
    "Transcriptome" = "mediumvioletred",
    "Proteome"= "darkslategrey",
    "Phosphoproteome" ="darkcyan", 
    "Secretome" = "darkseagreen3"
  )


pal_direction <-
  c(
    "down" = "dodgerblue4",
    "no direction" = "grey90",
    "up" = "indianred3"
  )



```

```{r}
secreted_proteins <- msigdbr::msigdbr(species = "Homo sapiens") %>% 
  filter(gs_id %in% c('M5889', 'M5885'))


myo_genes <- openxlsx::read.xlsx("data/41586_2020_2941_MOESM4_ESM.xlsx", sheet = "Human_PDGFRBplus_Level2_Specifi") %>% 
  separate(`X1`, into = c("ensembl", "Gene"), remove = T, sep = ";") %>% 
  separate(Gene, into = c("Gene", "iso"), sep = "\\.", remove = F) %>% 
  group_by(Gene) %>% 
  summarise_all(.funs = mean) %>%  select(Gene, Myofibroblasts) %>% 
  filter(Myofibroblasts > 0.2)
```


```{r}
# load("data/processed_data/2024-07-24_data_list.RData")
# load("data/processed_data/2024-07-24_diff_results.RData")
# 
#load("data/processed_data/2024-02-27_data_list.RData")
# load("data/processed_data/2024-02-27_diff_results.RData")

load("data/processed_data/2024-08-15_data_list.RData")
load("data/processed_data/2024-08-15_diff_results.RData")


```



# A

study design/overvuew
```{r}
diff_results %>%
  distinct(modality, feature_id) %>% 
  group_by(modality) %>% 
  count
```


# B 

images

# C

```{r}
load("results/2024-07-08_imaging_data_fig1.RData")

imaging_data$plot_data %>% 
  mutate(condition = str_replace(condition, "ctrl", "control")) %>% 
  mutate(condition = str_replace(condition, "TGF", "TGF-\u03b2")) %>% 
  ggplot(aes(x=time, y= norm_sqrt_mean_int, col=condition, fill = condition, group = condition)) + 
  geom_jitter(size = 3, alpha = 0.8, width = 0.8) +
  geom_smooth(method = "lm",se = T) +
  scale_color_manual(values = pal_condition) +
  scale_fill_manual(values = pal_condition) +
  labs(y = "COLI - norm fluorescence intensity", x = "treatment duration [h]")
```

## significance test

```{r}
emmeans::emmeans(
  imaging_data$model_data,
  ~ condition * time,
  adjust = "sidak",
  na.action = na.exclude
) %>%
  multcomp::cld(
    .,
    alpha = 0.05,
    Letter = letters,
    adjust = "sidak"
  )
```


# D venns

```{r}
library(VennDiagram)
```


```{r}
input <- diff_results %>%
  # only secreted
  filter(!(modality == "secretomics" & !(feature_id %in% secreted_proteins$gene_symbol))) %>% 
  #filter(modality != "phospho") %>%
  ##mutate(modality = ifelse(modality == "rna", "rna", "protoemics technologies")) %>%
  # modify feature_id to be able to collapse phospho
  mutate(feature_id = str_replace_all(feature_id, "___", "_")) %>%
  separate(feature_id, into = c("feature_id", "sequence", "multiplicity", "siteID")) %>%
   filter(modality != "phospho")%>%
  mutate(modality = ifelse(modality == "rna", "RNAseq", "Proteomics\ntechnologies")) %>% 
  ungroup() %>%
  distinct(modality, feature_id)

input_list <- split(input, input$modality)

p <- venn.diagram(
  x = list(
    input_list$RNAseq  %>% select(feature_id) %>% unlist(),
    input_list$`Proteomics
technologies` %>% select(feature_id) %>% unlist()
  ),
  filename = NULL,
  category.names = c("rna", "prot"),
  col = c("mediumseagreen", "#440154ff"),
  fill = c(alpha("mediumseagreen", 0.5), alpha("#440154ff", 0.5)),
  cex = 0.8,
  cex.prop = NULL,
  cat.cex = 1,
  cat.default.pos = "text",
  fontfamily = "serif",
  ext.percent = .005,
  # cat.pos = c(-27, 27),
  cat.dist = c(0.055, 0.055)
)
dev.off()
grid::grid.draw(p)
```

```{r}
input <- diff_results %>%
  # only secreted
  #filter(!(modality == "secretomics" & !(feature_id %in% secreted_proteins$gene_symbol))) %>% 
  # modify feature_id to be able to collapse phospho
  mutate(feature_id = str_replace_all(feature_id, "___", "_")) %>%
  separate(feature_id, into = c("feature_id", "sequence", "multiplicity", "siteID")) %>%
  ungroup() %>%
  filter(modality != "rna" & modality != "phospho")%>%
  distinct(modality, feature_id)

input_list <- split(input, input$modality)

p <- venn.diagram(
  x = list(
    input_list$input_phospho  %>% select(feature_id) %>% unlist(),
    input_list$proteomics %>% select(feature_id) %>% unlist(),
    input_list$secretomics %>% select(feature_id) %>% unlist()
  ),
  filename = NULL,
  category.names = c("phos", "prot", "sec"),
  col = c("mediumseagreen", "#440154ff", "darkcyan"),
  fill = c(alpha("mediumseagreen", 0.5), alpha("#440154ff", 0.5), alpha("darkcyan", 0.5)),
  cex = 0.8,
  cex.prop = NULL,
  cat.cex = 1,
  cat.default.pos = "text",
  fontfamily = "serif",
  ext.percent = .005,
  # cat.pos = c(-27, 27),
  cat.dist = c(0.055, 0.055, 0.055)
)
dev.off()
grid::grid.draw(p)
```

# E PCAs

```{r}
load("data/processed_data/2024-08-15_data_list.RData")
#load("data/processed_data/2024-02-27_data_list.RData")
pca_plot <- function(data_matrix, metadata, x, y, fx, fy, l, u) {
  m <- data_matrix 
  n_obs <- nrow(m)

  m <- m[matrixStats::rowVars(m, na.rm = T) %>%
    order(decreasing = T) %>%
    head((n_obs / 100) * 10), ]

  pca <- prcomp(t(m))
  
  data <- pca$x %>%
    as.data.frame() %>%
    rownames_to_column("matrix_id") %>%
    select(matrix_id, x = {{ x }}, y = {{ y }}) %>%
    inner_join(metadata) %>%
    mutate(group = paste0(condition, "_",time))
  
  var <- (pca$sdev)^2/sum(pca$sdev^2)*100 
  vx <- as.numeric(str_extract(as.character({{ x }}) , "\\d"))
  vy <- as.numeric(str_extract(as.character({{ y }}) , "\\d"))

  p <- data %>%
    ggplot(aes(x = x * fx, y = y * fy, fill = group, shape = condition)) +
    geom_hline(yintercept = 0, colour = "lightgrey") +
    geom_vline(xintercept = 0, colour = "lightgrey") +
    geom_point(size = 4, stroke = 1, colour = "grey60") +
    scale_fill_manual(values = pal) +
    scale_shape_manual(values = c(21, 24)) +
    cowplot::panel_border() +
    coord_fixed() +
    lims(x = c(l,u), y = c(l,u)) +
    guides(fill = "none", shape = "none") +
    labs(
      x = paste0(as.character({{ x }}), " (", round(var[vx], digits = 1), "%)"),
      y = paste0(as.character({{ y }}), " (", round(var[vy], digits = 1), "%)")
    )

  plot(p)
}


p <- cowplot::plot_grid(
  pca_plot(data_list$rna$data_matrix, data_list$rna$metadata, "PC1", "PC2", 1, -1, -30, 45),
  pca_plot(data_list$proteomics$data_matrix, data_list$proteomics$metadata, "PC1", "PC2", 1, 1, -10, 15),
  pca_plot(data_list$input_phospho$data_matrix, data_list$input_phospho$metadata, "PC2", "PC3", -1, 1, -15, 12),
  pca_plot(data_list$secretomics$data_matrix, data_list$secretomics$metadata, "PC1", "PC2", 1, 1, -12, 10),
  nrow = 1
)
plot(p)
  
```

# F line plots

```{r, fig.width=12}
lfc_cutoff <- log2(1.5)
p_cutoff <- 0.05

hit_table <- diff_results %>%
  filter(!(modality == "secretomics" & !(feature_id %in% secreted_proteins$gene_symbol))) %>% 
  subset(abs(logFC) >= lfc_cutoff & adj.P.Val <= p_cutoff) %>%
   filter(!(modality == "rna" & abs(logFC) < log2(2))) %>% 
  mutate(status = ifelse(logFC > 0, "up", "down")) %>% 
  group_by(modality, comparison, status, time) %>%
  summarise(count = n())

dummy_df <- data.frame(modality = c(unique(hit_table$modality)),
                       time = factor(c("0h")), 
                       count = 0,
                       status = c(rep("up", length(unique(hit_table$modality))), rep("down", length(unique(hit_table$modality)))))

df<- hit_table %>%
  bind_rows(dummy_df) %>%
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h"))) %>%
  ungroup() %>% 
  tidyr::complete(time, status, modality, fill = list(count = 0)) %>% 
  mutate(time = as.numeric(str_remove(time,"h"))) %>% 
  filter(modality != "phospho") %>% 
  mutate(modality = ifelse(modality == "rna", "Transcriptome", modality)) %>% 
   mutate(modality = ifelse(modality == "input_phospho", "Phosphoproteome", modality)) %>% 
  mutate(modality = ifelse(modality == "proteomics", "Proteome", modality)) %>% 
  mutate(modality = ifelse(modality == "secretomics", "Secretome", modality)) %>% 
   mutate(data = "all") 

hit_table <- diff_results %>%
   filter(!(modality == "secretomics" & !(feature_id %in% secreted_proteins$gene_symbol))) %>% 
  mutate(feature_id = str_replace_all(feature_id, "___", "_")) %>%
  separate(feature_id, into = c("feature_id", "sequence", "multiplicity", "siteID")) %>%
   filter(feature_id %in% myo_genes$Gene) %>% 
  subset(abs(logFC) >= lfc_cutoff & adj.P.Val <= p_cutoff) %>%
   filter(!(modality == "rna" & abs(logFC) < log2(2))) %>% 
  mutate(status = ifelse(logFC > 0, "up", "down")) %>% 
  group_by(modality, comparison, status, time) %>%
  summarise(count = n())

dummy_df <- data.frame(modality = c(unique(hit_table$modality)),
                       time = factor(c("0h")), 
                       count = 0,
                       status = c(rep("up", length(unique(hit_table$modality))), rep("down", length(unique(hit_table$modality)))))

df_myo<- hit_table %>%
  bind_rows(dummy_df) %>%
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h"))) %>%
  ungroup() %>% 
  tidyr::complete(time, status, modality, fill = list(count = 0)) %>% 
  mutate(time = as.numeric(str_remove(time,"h"))) %>% 
  filter(modality != "phospho") %>% 
  mutate(modality = ifelse(modality == "rna", "Transcriptome", modality)) %>% 
   mutate(modality = ifelse(modality == "input_phospho", "Phosphoproteome", modality)) %>% 
  mutate(modality = ifelse(modality == "proteomics", "Proteome", modality)) %>% 
  mutate(modality = ifelse(modality == "secretomics", "Secretome", modality)) %>% 
   mutate(data = "myofibroblast specific genes") 


df_myo %>% 
  bind_rows(df) %>% 
  mutate(modality = factor(modality, levels = c("Transcriptome", "Proteome", "Phosphoproteome", "Secretome"))) %>% 
  ggplot(aes(x = time, y = count, colour = status, linetype = data)) +
  geom_point(size = 2.5) +
  geom_line(aes(group = paste0(status, data)), size = 1) +
  scale_colour_manual(values = c("up" = "indianred", "down"= "dodgerblue4")) +
  #guides(x = guide_axis(angle = 60)) +
  facet_wrap(~modality, scales = "free_y", nrow =1) +
  labs(x = "time [h]", y= "# of hits") +
  guides(colour = F, linetype = F) +
  cowplot::panel_border() +
  theme(strip.background = element_blank(), strip.text = element_blank())+ 
  theme(panel.spacing = unit(2.1, "lines"))

```


```{r}

df_myo %>% 
  bind_rows(df) %>% 
  filter(modality == "Transcriptome" & data != "all") %>% 
  ggplot(aes(x = time, y = count, colour = status)) +
  geom_point(size = 2.5) +
  geom_line(aes(group = paste0(status, data)), size = 1, linetype = 2) +
  scale_colour_manual(values = c("up" = "indianred", "down"= "dodgerblue4")) +
  #guides(x = guide_axis(angle = 60)) +
  facet_wrap(~modality, scales = "free_y", nrow =1) +
  labs(x = "time [h]", y= "# of hits") +
  guides(colour = F, linetype = F) +
  cowplot::panel_border() +
  theme(strip.background = element_blank(), strip.text = element_blank())+ 
  theme(panel.spacing = unit(2.1, "lines"))
```


# Supp

## A corrs

### rna

```{r}
library(ComplexHeatmap)
m <- data_list$rna$data_matrix %>%
  t() %>%
  as.data.frame() %>%
  rownames_to_column("matrix_id") %>%
  inner_join(data_list$rna$metadata %>%  select(matrix_id, condition, time, replicate)) %>%
  mutate(group = paste0(condition, "_", time, "_", replicate)) %>%
  select(-matrix_id, -condition, -time, -replicate) %>% 
  column_to_rownames("group") %>% 
  as.matrix() %>% 
  t() %>% 
  cor()

row_anno <- ComplexHeatmap::rowAnnotation(
  #condition = str_extract(rownames(m), "TGF|ctrl"),
  #replicate = str_extract(rownames(m), "A|B|C"),
  group = str_remove(rownames(m),"_A|_B|_C|_D"),
  col = list(group = pal_cond_time)
  )

col_anno <- ComplexHeatmap::HeatmapAnnotation(
  #condition = str_extract(rownames(m), "TGF|ctrl"),
  #replicate = str_extract(colnames(m), "A|B|C"),
  group = str_remove(colnames(m),"_A|_B|_C|_D"),
  col = list(group = pal_cond_time)
  )

m %>% 
  ComplexHeatmap::Heatmap(
      col = c("white", "indianred", "darkred"), rect_gp = gpar(type = "none"), name = "Pearson\ncorrelation", left_annotation = row_anno, show_row_names = F, bottom_annotation = col_anno, show_column_names = F, show_column_dend = F,
      cell_fun = function(j, i, x, y, w, h, fill) {
        if(as.numeric(x) <= 1 - as.numeric(y) + 1e-6) {
            grid.rect(x, y, w, h, gp = gpar(fill = fill, col = fill))
        }
    }
  )
```

### proteomics
```{r}
library(ComplexHeatmap)
m <- data_df %>%  
  filter(id == "proteomics") %>% 
  #mutate(group = str_replace(group, "ctrl", "control")) %>% 
  acast(feature_id ~group, value.var = "value") %>% 
  cor()

row_anno <- ComplexHeatmap::rowAnnotation(
  #condition = str_extract(rownames(m), "TGF|ctrl"),
  #replicate = str_extract(rownames(m), "A|B|C"),
  group = str_remove(rownames(m),"_A|_B|_C|_D"),
  col = list(group = pal_cond_time)
  )

col_anno <- ComplexHeatmap::HeatmapAnnotation(
  #condition = str_extract(rownames(m), "TGF|ctrl"),
  #replicate = str_extract(colnames(m), "A|B|C"),
  group = str_remove(colnames(m),"_A|_B|_C|_D"),
  col = list(group = pal_cond_time)
  )

m %>% 
  ComplexHeatmap::Heatmap(
      col = c("white", "indianred", "darkred"), rect_gp = gpar(type = "none"), name = "Pearson\ncorrelation", left_annotation = row_anno, show_row_names = F, bottom_annotation = col_anno, show_column_names = F, show_column_dend = F,
      cell_fun = function(j, i, x, y, w, h, fill) {
        if(as.numeric(x) <= 1 - as.numeric(y) + 1e-6) {
            grid.rect(x, y, w, h, gp = gpar(fill = fill, col = fill))
        }
    }
  )
```


### phospho
```{r}

m <- data_df %>%  
  filter(id == "input_phospho") %>% 
  #mutate(group = str_replace(group, "ctrl", "control")) %>% 
  acast(feature_id ~group, value.var = "value") %>% 
  cor()

row_anno <- ComplexHeatmap::rowAnnotation(
  #condition = str_extract(rownames(m), "TGF|ctrl"),
  #replicate = str_extract(rownames(m), "A|B|C"),
  group = str_remove(rownames(m),"_A|_B|_C|_D"),
  col = list(group = pal_cond_time)
  )

col_anno <- ComplexHeatmap::HeatmapAnnotation(
  #condition = str_extract(rownames(m), "TGF|ctrl"),
  #replicate = str_extract(colnames(m), "A|B|C"),
  group = str_remove(colnames(m),"_A|_B|_C|_D"),
  col = list(group = pal_cond_time)
  )

m %>% 
  ComplexHeatmap::Heatmap(
      col = c("white", "indianred", "darkred"),
      rect_gp = gpar(type = "none"),
      name = "Pearson\ncorrelation", 
      left_annotation = row_anno,
      show_row_names = F,
      bottom_annotation = col_anno, 
      show_column_names = F,
      show_column_dend = F, 
      cell_fun = function(j, i, x, y, w, h, fill) {
        if(as.numeric(x) <= 1 - as.numeric(y) + 1e-6) {
            grid.rect(x, y, w, h, gp = gpar(fill = fill, col = fill))
        }
    }
  )
```

### secretomics

```{r}

m <- data_df %>%  
  filter(id == "secretomics") %>% 
  #mutate(group = str_replace(group, "ctrl", "control")) %>% 
  acast(feature_id ~group, value.var = "value") %>% 
  cor()

row_anno <- ComplexHeatmap::rowAnnotation(
  #condition = str_extract(rownames(m), "TGF|ctrl"),
  #replicate = str_extract(rownames(m), "A|B|C"),
  group = str_remove(rownames(m),"_A|_B|_C"),
  col = list(group = pal_cond_time)
  )

col_anno <- ComplexHeatmap::HeatmapAnnotation(
  #condition = str_extract(rownames(m), "TGF|ctrl"),
  #replicate = str_extract(colnames(m), "A|B|C"),
  group = str_remove(colnames(m),"_A|_B|_C"),
  col = list(group = pal_cond_time)
  )

m %>% 
  ComplexHeatmap::Heatmap(
      col = c("white", "indianred", "darkred"),
      rect_gp = gpar(type = "none"),
      name = "Pearson\ncorrelation", 
      left_annotation = row_anno,
      show_row_names = F,
      bottom_annotation = col_anno, 
      show_column_names = F,
      show_column_dend = F, 
      cell_fun = function(j, i, x, y, w, h, fill) {
        if(as.numeric(x) <= 1 - as.numeric(y) + 1e-6) {
            grid.rect(x, y, w, h, gp = gpar(fill = fill, col = fill))
        }
    }
  )
```

## B phospho PC1 PC2

```{r}
data_pca %>% 
  filter(id == "input_phospho") %>% 
  ggplot(aes(x = PC1, y = PC2, fill = group, shape = condition)) +
  geom_hline(yintercept = 0, colour = "lightgrey") +
  geom_vline(xintercept = 0, colour = "lightgrey") +
   geom_point(size = 4, stroke = 1, colour = "grey60") +
  scale_fill_manual(values =pal)+
   scale_shape_manual(values = c(21, 24)) +
  cowplot::panel_border()  +
  lims(x = c(-42,42), y = c(-18,18)) +
  guides(fill = "none", shape = "none") +
  labs(x = "",  y = "")
  labs(subtitle = "Phosphoproteome")
```


## C volcanos

```{r, fig.width = 14, fig.height=6}
# volcano plot
diff_results %>% 
  mutate(hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>% 
  mutate(hit = ifelse(adj.P.Val < 0.05 & abs(logFC) < log2(2) & modality == "rna", "no hit", hit)) %>% 
  mutate(status = ifelse(hit == "hit" & logFC > 0, "up", "not affected")) %>% 
  mutate(status = ifelse(hit == "hit" & logFC < 0, "down", status)) %>% 
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h"))) %>% 
  mutate(modality = ifelse(modality == "rna", "Transcriptome", modality)) %>% 
   mutate(modality = ifelse(modality == "input_phospho", "Phosphoproteome", modality)) %>% 
  mutate(modality = ifelse(modality == "proteomics", "Proteome", modality)) %>% 
  mutate(modality = ifelse(modality == "secretomics", "Secretome", modality)) %>% 
  filter(modality != "phospho") %>% 
  ggplot(aes(x = logFC, y = -log10(P.Value), color = status)) +
  #scale_color_gradient2(low = "dodgerblue4", mid = "lightgrey", high = "indianred", midpoint = 0) +
  geom_hline(yintercept = 0, lty = 1) +
  geom_vline(xintercept = c(log2(1/1.5), log2(1.5)), lty = 2) +
  geom_point(alpha = 0.5) +
  facet_grid(rows = vars(modality), cols = vars(time)) +
  scale_colour_manual(values = c("up" = "indianred","not affected" ="lightgrey",  "down"= "dodgerblue4")) +
  cowplot::theme_cowplot()+
  cowplot::panel_border() +
  labs(x = "log2 fold-change TGFbeta/ctrl")
```


## D Myo histogram

```{r}
data <- openxlsx::read.xlsx("data/41586_2020_2941_MOESM4_ESM.xlsx", sheet = "Human_PDGFRBplus_Level2_Specifi") %>% 
  separate(`X1`, into = c("ensembl", "Gene"), remove = T, sep = ";") %>% 
  separate(Gene, into = c("Gene", "iso"), sep = "\\.", remove = F) %>% 
  group_by(Gene) %>% 
  summarise_all(.funs = mean) %>%  select(Gene, Myofibroblasts)  

data %>% 
  ggplot(aes(x = Myofibroblasts)) +
  geom_histogram() +
  geom_vline(xintercept = 0.2) +
  labs(y = "number of genes", x = "Myofibroblast specificity score")

```


## E correlation

```{r}
hits <- diff_results %>% 
  mutate(hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>% 
  mutate(hit = ifelse(adj.P.Val < 0.05 & abs(logFC) < log2(2) & modality == "rna", "no hit", hit)) %>% 
  filter(hit == "hit")

col_fun = circlize::colorRamp2(c(-0.9,-0.4, 0, 0.4, 0.9), c("dodgerblue4","lightblue", "grey95", "indianred", "darkred"))
m <- diff_results %>% 
  filter(modality != "phospho" & modality != "input_phospho") %>% 
  filter(feature_id %in% hits$feature_id) %>% 
  dcast(feature_id ~ modality + time, value.var = "t", fill = NA) %>% 
  drop_na() %>% 
  column_to_rownames("feature_id") %>% 
  as.matrix() %>% 
  cor(method = "pearson")

# m %>%  
#   ComplexHeatmap::Heatmap(
#     col = col_fun, 
#     cluster_columns = F,
#     cluster_rows = F,
#     #row_km = 2,
#     rect_gp = gpar(lwd=2, col = "white"),
#     name = "Pearson\ncorrelation"
#     )

corrplot::corrplot(m,type = "upper",  method = "color",  col.lim = c(-1, 1),addCoef.col = 'white', number.cex = 0.65,tl.pos = "n", number.digits = 1,number.font = 1,
                        col = colorRampPalette(c("dodgerblue4","lightblue", "grey95", "indianred", "darkred"))(10),addgrid.col = 'white')
```


```{r}
hits <- diff_results %>% 
  mutate(hit = ifelse(adj.P.Val < 0.05 & abs(logFC) > log2(1.5), "hit", "no hit")) %>% 
  mutate(hit = ifelse(adj.P.Val < 0.05 & abs(logFC) < log2(2) & modality == "rna", "no hit", hit)) %>% 
  filter(hit == "hit")

col_fun = circlize::colorRamp2(c(-0.9,-0.4, 0, 0.4, 0.9), c("dodgerblue4","lightblue", "grey95", "indianred", "darkred"))
m <- diff_results %>% 
  filter(modality == "input_phospho") %>% 
  filter(feature_id %in% hits$feature_id) %>% 
  dcast(feature_id ~ modality + time, value.var = "t", fill = NA) %>% 
  drop_na() %>% 
  column_to_rownames("feature_id") %>% 
  as.matrix() %>% 
  cor(method = "pearson")
# m %>%  
#   ComplexHeatmap::Heatmap(
#     col = col_fun, 
#     cluster_columns = F,
#     cluster_rows = F,
#     #row_km = 2,
#     rect_gp = gpar(lwd=2, col = "white"),
#     name = "Pearson\ncorrelation"
#     )

corrplot::corrplot(m,type = "upper",  method = "color",  col.lim = c(-1, 1),addCoef.col = 'white', number.cex = 0.65,tl.pos = "n", number.digits = 1,number.font = 1,
                        col = colorRampPalette(c("dodgerblue4","lightblue", "grey95", "indianred", "darkred"))(10),addgrid.col = 'white')
```

```{r, fig.width=12}
lfc_cutoff <- log2(1.5)
p_cutoff <- 0.05

hit_table <- diff_results %>%
  filter(!(modality == "secretomics" & !(feature_id %in% secreted_proteins$gene_symbol))) %>% 
  subset(abs(logFC) >= lfc_cutoff & adj.P.Val <= p_cutoff) %>%
   filter(!(modality == "rna" & abs(logFC) < log2(2))) %>% 
  mutate(status = ifelse(logFC > 0, "up", "down")) %>% 
  group_by(modality, comparison, status, time) %>%
  summarise(count = n())

dummy_df <- data.frame(modality = c(unique(hit_table$modality)),
                       time = factor(c("0h")), 
                       count = 0,
                       status = c(rep("up", length(unique(hit_table$modality))), rep("down", length(unique(hit_table$modality)))))

df<- hit_table %>%
  bind_rows(dummy_df) %>%
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h"))) %>%
  ungroup() %>% 
  tidyr::complete(time, status, modality, fill = list(count = 0)) %>% 
  mutate(time = as.numeric(str_remove(time,"h"))) %>% 
  filter(modality != "phospho") %>% 
  mutate(modality = ifelse(modality == "rna", "Transcriptome", modality)) %>% 
   mutate(modality = ifelse(modality == "input_phospho", "Phosphoproteome", modality)) %>% 
  mutate(modality = ifelse(modality == "proteomics", "Proteome", modality)) %>% 
  mutate(modality = ifelse(modality == "secretomics", "Secretome", modality)) %>% 
   mutate(data = "all") 

hit_table <- diff_results %>%
   filter(!(modality == "secretomics" & !(feature_id %in% secreted_proteins$gene_symbol))) %>% 
  mutate(feature_id = str_replace_all(feature_id, "___", "_")) %>%
  separate(feature_id, into = c("feature_id", "sequence", "multiplicity", "siteID")) %>%
   filter(feature_id %in% myo_genes$Gene) %>% 
  subset(abs(logFC) >= lfc_cutoff & adj.P.Val <= p_cutoff) %>%
   filter(!(modality == "rna" & abs(logFC) < log2(2))) %>% 
  mutate(status = ifelse(logFC > 0, "up", "down")) %>% 
  group_by(modality, comparison, status, time) %>%
  summarise(count = n())

dummy_df <- data.frame(modality = c(unique(hit_table$modality)),
                       time = factor(c("0h")), 
                       count = 0,
                       status = c(rep("up", length(unique(hit_table$modality))), rep("down", length(unique(hit_table$modality)))))

df_myo<- hit_table %>%
  bind_rows(dummy_df) %>%
  mutate(time = factor(time, levels = c("0h", "0.08h", "1h", "12h","24h", "48h", "72h", "96h"))) %>%
  ungroup() %>% 
  tidyr::complete(time, status, modality, fill = list(count = 0)) %>% 
  mutate(time = as.numeric(str_remove(time,"h"))) %>% 
  filter(modality != "phospho") %>% 
  mutate(modality = ifelse(modality == "rna", "Transcriptome", modality)) %>% 
   mutate(modality = ifelse(modality == "input_phospho", "Phosphoproteome", modality)) %>% 
  mutate(modality = ifelse(modality == "proteomics", "Proteome", modality)) %>% 
  mutate(modality = ifelse(modality == "secretomics", "Secretome", modality)) %>% 
   mutate(data = "myofibroblast specific genes") 


tt <- df_myo %>% 
  bind_rows(df) 
```
